assets/async/point.js

```diff
@@ -14,36 +16,69 @@ const pointSelect = ({ runConfiguration }) => {
-    resync(store, pointSelect, ({ point }, io, dispatch) => {
+    resync(store, pointSelect, (state, io, dispatch, previousState) => {
+        let { point } = state
         let pointIsValid = point !== null && point.x && point.y
 
-        dispatch(setElevation(null))
-
         if (pointIsValid) {
-            let region = findClosestRegion(point.x, point.y)
-
-            let url = '/arcgis/rest/services/' + region.name + '_dem/MapServer/identify/?' + urlEncode({
-                f: 'json',
-                tolerance: '2',
-                imageDisplay: '1600,1031,96',
-                geometryType: 'esriGeometryPoint',
-                mapExtent: '0,0,0,0',
-                geometry: JSON.stringify({x: point.x, y: point.y})
+
+            dispatch(setElevation(null))
+            dispatch(requestRegions())
+            let regionUrl = '/sst/regions/?' + urlEncode({
+                point: point.x + ',' + point.y
             })
 
-            io.get(url).then(response => response.json()).then(json => {
+            io.get(regionUrl).then(response => response.json()).then(json => {
                 let results = json.results
-                let value = null
+                let validRegions = results.map((a) => {
+                    return a.name
+                })
+
+                dispatch(receiveRegions(validRegions))  // Always update valid regions
 
-                if (results.length) {
-                    value = results[0].attributes['Pixel value']
+                let region = ''
+                if (validRegions.length) {
+                    region = validRegions[0]
                 }
 
-                if (isNaN(value)) {
-                    value = null
+                if (store.getState().runConfiguration.regionMethod === 'auto') {
+                    dispatch(setRegion(region))
                 }
+                return region
+            }).then(region => {
+
+                let url = '/arcgis/rest/services/' + region + '_dem/MapServer/identify/?' + urlEncode({
+                        f: 'json',
+                        tolerance: '2',
+                        imageDisplay: '1600,1031,96',
+                        geometryType: 'esriGeometryPoint',
+                        mapExtent: '0,0,0,0',
+                        geometry: JSON.stringify({x: point.x, y: point.y})
+                    })
+
+                io.get(url).then(response => response.json()).then(json => {
+                    let results = json.results
+                    let value = null
 
-                dispatch(setElevation(value))
+                    if (results.length) {
+                        value = results[0].attributes['Pixel value']
+                    }
+
+                    if (isNaN(value)) {
+                        value = null
+                    }
+
+                    dispatch(setElevation(value))
+                })
+
+                let requests = fetchValues(store, state, io, dispatch, previousState, region)
```

Comment:
The issue I had with that approach was that whenever I clicked on a point, the values wouldn't load unless I added a new variable, which would then trigger the actions to get the right values. I'll give it another pass and see if I can get that working so that this isn't necessary.